docs_src/microservices/device/profile/V2DeviceProfileRef.md

```diff
@@ -0,0 +1,65 @@
+# V2 Device Profile Reference
+
+This chapter details the structure of a Device Profile and allowable values for
+its fields.
+
+Device Profile
+--------------
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+name | String | Y | Must be unique in the EdgeX deployment
+manufacturer | String | N |
+model | String | N |
+labels | Array of String | N |
+deviceResources | Array of DeviceResource | Y |
+deviceCommands  | Array of DeviceCommand | N |
+
+DeviceResource
+--------------
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+name | String | Y | Must be unique in the EdgeX deployment
+description | String | N |
+isHidden | Bool | N | Expose the DeviceResource to Command Service or not, default false
+tag | String | N |
+attributes | String-Interface Map | N | Each Device Service should define required and optional keys
+properties | ResourceProperties | Y |
+
+ResourceProperties
+---------------
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+valueType | Enum | Y | `Uint8`, `Uint16`, `Uint32`, `Uint64`, `Int8`, `Int16`, `Int32`, `Int64`, `Float32`, `Float64`, `Bool`, `String`, `Binary`, `Uint8Array`, `Uint16Array`, `Uint32Array`, `Uint64Array`, `Int8Array`, `Int16Array`, `Int32Array`, `Int64Array`, `Float32Array`, `Float64Array`, `BoolArray`
+readWrite | Enum | Y | `R`, `W`, `RW` 
+units | String | N |
+minimum | String | N |
+maximum | String | N |
+defaultValue | String | N | If present, should be compatible with the Type field
+mask | String | N | Only valid where Type is one of the integer types
+shift | String | N | Only valid where Type is one of the integer types
+scale | String | N | Only valid where Type is one of the integer or float types
+offset | String | N | Only valid where Type is one of the integer or float types
+base | String | N | Only valid where Type is one of the integer or float types
+assertion | String | N |
+mediaType | String | N |
+
+DeviceCommand
+-------------
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+name | String | Y | Must be unique in this profile. May have the same name as a DeviceResource but this will make the DeviceResource not accessible individually. Such a configuration is only recommended if Mappings are used: see below.
```

Comment:
Do we need to make it clear that command for a signal resource is redundant to just using that resource. i.e. only need Device Commands for those that operate on multiple device resources?

Reply:
I still think we need a note here to highlight that a device command with a single resource is redundant unless renaming and/or restricting R/W access. i.e resource is RW, but device command is only Read.

Reply:
updated this in DeviceProfile page